src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -97,41 +96,87 @@ Json::Value doTx (RPC::Context& context)
     if (!isHexTxID (txid))
         return rpcError (rpcNOT_IMPL);
 
-    auto txn = context.app.getMasterTransaction ().fetch (
-        from_hex_text<uint256>(txid), true);
+    ClosedInterval<uint32_t> range;
 
-    if (!txn)
-        return rpcError (rpcTXN_NOT_FOUND);
+    auto rangeProvided = context.params.isMember (jss::min_ledger) &&
+        context.params.isMember (jss::max_ledger);
 
-    Json::Value ret = txn->getJson (JsonOptions::include_date, binary);
+    if (rangeProvided)
+    {
+        try
+        {
+            auto const& min = context.params[jss::min_ledger].asUInt ();
+            auto const& max = context.params[jss::max_ledger].asUInt ();
+
+            constexpr uint16_t MAX_RANGE = 1000;
+
+            if (max < min)
+                return rpcError (rpcINVALID_LGR_RANGE);
+
+            if (max - min > MAX_RANGE)
+                return rpcError (rpcEXCESSIVE_LGR_RANGE);
+
+            range = ClosedInterval<uint32_t> (min, max);
+        }
+        catch (...)
+        {
+            // One of the calls to `asUInt ()` failed.
+            return rpcError (rpcINVALID_LGR_RANGE);
+        }
+    }
+
+    auto ec = rpcSUCCESS;
+    Transaction::variant v;
```

Comment:
Instead of a variant here, how about a `Transaction::pointer`. Then in the `rangeProvided` branch, handle the `bool` alternative of the variant and return there, or set the pointer and continue. The non`rangeProvided` branch would just set the pointer.

Doing it this way probably require duplicating two lines of code, but I think that's an acceptable trade-off for the clearer code.

Reply:
Done.